velox/exec/tests/TableWriteTest.cpp

```diff
@@ -917,6 +917,61 @@ class TableWriteTest : public HiveConnectorTestBase {
   core::PlanNodeId tableWriteNodeId_;
 };
 
+class BasicTableWriteTest : public HiveConnectorTestBase {};
+
+TEST_F(BasicTableWriteTest, roundTrip) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(size, [](auto row) { return row; }),
+      makeFlatVector<int32_t>(
+          size, [](auto row) { return row * 2; }, nullEvery(7)),
+  });
+
+  auto sourceFilePath = TempFilePath::create();
+  writeToFile(sourceFilePath->path, data);
+
+  auto targetDirectoryPath = TempDirectoryPath::create();
+
+  auto rowType = asRowType(data->type());
+  auto plan = PlanBuilder()
+                  .tableScan(rowType)
+                  .tableWrite(targetDirectoryPath->path)
+                  .planNode();
+
+  auto results = AssertQueryBuilder(plan)
+                     .split(makeHiveConnectorSplit(sourceFilePath->path))
+                     .copyResults(pool());
+  ASSERT_EQ(2, results->size());
+
+  // First column has number of rows written in the first row and nulls in other
+  // rows.
+  auto rowCount = results->childAt(0)->as<FlatVector<int64_t>>();
```

Comment:
Consider using the constants `kFragmentChannel` inside `TableWriter.h` for better readability